### ISSUE TITLE
linter: Eslint plugin license header is not compatible with oxlint

### ISSUE THREAD

**Issue by rajzik** [oxc-project/oxc#18700]
### What version of Oxlint are you using?

1.42.0

### What command did you run?

`oxlint`

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "jsPlugins": ["eslint-plugin-license-header"],
  "rules": {
     "license-header/header": ["error", [
        "/* my header */"
     ]],
  }
}
```

### What happened?

License header is not detected even when present in the files.

This plugin was patched with patch package and diff:
```
diff --git a/node_modules/eslint-plugin-license-header/lib/rules/header.js b/node_modules/eslint-plugin-license-header/lib/rules/header.js
index 5e59ab3..58f8b7b 100644
--- a/node_modules/eslint-plugin-license-header/lib/rules/header.js
+++ b/node_modules/eslint-plugin-license-header/lib/rules/header.js
@@ -17,7 +17,7 @@ module.exports = {
       category: 'Possible Errors',
       recommended: true
     },
-    fixable: true,
+    fixable: 'code',
     schema: [
       {
         type: [ 'string', 'array' ],

```

**Comment by connorshea** [comment]
Err, what do you mean it was "patched"? Like your project patches it?

**Comment by camc314** [comment]
<img width="852" height="172" alt="Image" src="https://github.com/user-attachments/assets/98613153-6011-4004-b663-fbe621201324" />

This is a bug in the plugin rather than in oxlint - eslint requires that this property is either: undefined, `"code"` or `"whitespace"`, `true` isn't a valid option here.

<img width="1221" height="341" alt="Image" src="https://github.com/user-attachments/assets/b293708c-5082-4e10-ba15-06686fe7f5c5" />

The type has been set in eslint as code/whitespace strings (bool not allowed) since at least 2019


@overlookmotel I really don't want to have us handle this case, but what do you think?

**Comment by overlookmotel** [comment]
#18011 is the PR to fix it, if we decide we should.

**Comment by overlookmotel** [comment]
Similar to #17967 - that issue requests support for `false`.

**Comment by rajzik** [comment]
The issue is not fixable: true, this is something we patched in our codebase to make it running. After that the plugin doesn't work as expected, it does not add license header to the code and detects that licence header is not present even when it is present

**Comment by camc314** [comment]
> The issue is not fixable: true, this is something we patched in our codebase to make it running. After that the plugin doesn't work as expected, it does not add license header to the code and detects that licence header is not present even when it is present


Can you make a minimal repro for us to debug?

I tried to reproduce this here and it was working as expected:

https://github.com/camc314/oxlint-repros/tree/main/issue-18700

**Comment by camc314** [comment]
We've fixed the issue that means that the rule requires a patch.

Please re-open if you have a minimal repro for the plugin not working correctly - thanks
